src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -117,12 +117,15 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
 
             if (_host != null)
             {
+                bool isHostTypeIPv6 = _host.Split(':').Length - 1 > 1;
```

Comment:
Should it be just `_host.Contains(':')`? Why does it check for 2+ `:`?

Reply:
A valid IPv6 address contains at least two `:`, so I check it for two times.

Reply:
Split allocates unnecessarily - even if just per ConnectionPool, what about using `IndexOf` on the string twice?

Reply:
if it's performance critical, then I think more efficient way is:

```c#
var count = 0;
var isHostTypeIPv6 = false;
foreach (char c in _host)
{
  if (c == ':' && ++count > 1)
  {
    isHostTypeIPv6 = true;
    break;
  }
}
```

Reply:
Consider `host` is not a long string, `IndexOf` can make the code cleaner.